### ISSUE TITLE
NullPointerException on MutableAudioFrame#store() in 1.3.78

### ISSUE THREAD

**Issue by sreid8** [sedmelluq/lavaplayer#680]
Similar to closed issue #168 which was closed with advice to move to 1.3.16 and no other context, when following https://docs.discord4j.com/music-bot-tutorial/ using discord4j version 3.2.1 and lavaplayer version 1.3.78.

Unlike #168, the audio _never_ starts playing. This has been tested with multiple sources (youtube link to _Never Gonna Give You Up_, local 320kbps MP3, and local 128kbps re-encode MP3 file) and the same happens, it just takes longer to start happening with the 320kbps file.

I can provide more info including code samples as requested. I'm not doing that now only so that this text doesn't get longer than it needs to be.

When executing `val didProvide = player.provide(frame)` in LavaPlayerAudioProvider, the following stack trace occurs:
```
java.lang.NullPointerException: null
	at com.sedmelluq.discord.lavaplayer.track.playback.MutableAudioFrame.store(MutableAudioFrame.java:32)
	at com.sedmelluq.discord.lavaplayer.track.playback.NonAllocatingAudioFrameBuffer.popFrame(NonAllocatingAudioFrameBuffer.java:183)
	at com.sedmelluq.discord.lavaplayer.track.playback.NonAllocatingAudioFrameBuffer.provide(NonAllocatingAudioFrameBuffer.java:136)
	at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.provide(LocalAudioTrackExecutor.java:440)
	at com.sedmelluq.discord.lavaplayer.track.BaseAudioTrack.provide(BaseAudioTrack.java:101)
	at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayer.provide(DefaultAudioPlayer.java:232)
	at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayer.provide(DefaultAudioPlayer.java:212)
	at com.sreid.discord.voicebot.audio.LavaPlayerAudioProvider.provide(LavaPlayerAudioProvider.kt:20)
	at discord4j.voice.LocalVoiceSendTask.run(LocalVoiceSendTask.java:63)
	at reactor.core.scheduler.PeriodicSchedulerTask.call(PeriodicSchedulerTask.java:49)
	at reactor.core.scheduler.PeriodicSchedulerTask.run(PeriodicSchedulerTask.java:63)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by sreid8** [comment]
Additional context: tested with both jdk 1.8 and jdk 11 on MacOS 12.1 and tested on jdk 11 on Ubuntu 20.04LTS and saw the same behavior.

**Comment by sreid8** [comment]
Closing this issue for a few reasons I'll explain below:

1. 50% of this was user error; I'm writing in Kotlin and I made a mistake in my translation from the Java examples in the tutorial doc. I forgot to set the buffer in the `LavaPlayerAudioProvider`, for posterity, my current impl looks like: 
```kotlin
class LavaPlayerAudioProvider(private val player: AudioPlayer) :
  AudioProvider(
    ByteBuffer.allocate(StandardAudioDataFormats.DISCORD_OPUS.maximumChunkSize())) {
  
  private val frame = MutableAudioFrame()

  init {
    this.frame.setBuffer(buffer)
  }

  override fun provide(): Boolean {
    val didProvide = player.provide(frame)

    if (didProvide) {
      (buffer as Buffer).flip()
    }
    return didProvide
  }
}
```
My original implementation didn't have the `init` block.

2. I'm using a Mac, and I'm using Adoptopenjdk. This shouldn't be an issue, but apparently it is. Even once I'd fixed the issue above, Adoptopenjdk 8 apparently has a backport of `ByteBuffer` containing its own implementations of `flip()` and `clear()`, which caused `NoSuchMethod` exceptions when they were expected to be on `Buffer` and not `ByteBuffer` by 3.2.1 of Discord4j that's in Maven Central, which, I assume, was compiled with a more standard jdk8. The exception I saw was: `java.lang.NoSuchMethodException: java.nio.ByteBuffer.clear()ljava/nio/ByteBuffer;`

I add point 2 not because it's LavaPlayer's problem at all, but that I assume a lot of people will get here via the tutorial I got here through and I'm hoping I can put enough keywords in that quick write up that someone in the future will do what I did and just move over to a different jdk. In my case, I ran my code on Openjdk8 on Ubuntu and it worked like a charm.
